Added new readme file (#1081)
stackrox/scanner: scripts/node/README.md

# Steps to benchmark node scanning 

## Pre steps:

> 1.Make sure docker is installed on your local laptop

Reviewer: ```suggestion
1. Make sure docker is installed on your local laptop
```
Thank you for converting to Markdown! Great to see the renderer picking it up.
Looking at [the rendered file](https://github.com/stackrox/scanner/blob/01bfda2267e65abe911dd949b49d0c93022279b0/scripts/node/README.md), it seems that the renderer didn't pick up the ordered list.
I suspect this is because there are missing whitespaces between the number and the text ([see also: Spec](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax#lists)). See my change for an example. 

I recommend this to be done for all list items in this file.
Maybe we should also capitalize the first word of each list item.

2.execute
```

> docker run -it --rm registry.access.redhat.com/ubi8/ubi

Reviewer: A docker run alone might work now, but only because the image has a default entrypoint which is bash.
How about we make this more explicit?
```suggestion
docker run -it --rm --entrypoint /bin/bash registry.access.redhat.com/ubi8/ubi
```

```
3.make sure **Vim, RPM and RPM-build** are installed in the running ubi8 bash. If not, do yum search <package name> and yum install <package name>

---------------------------------------------------------------------------------------------

(optional) create a temporary/target folder in the running ubi8 bash for RPM specs. e.g mkdir temp-specs

> 1.go to the target folder and create a python file to generate RPM specs (see the generate-junit-reports.sh in scanner/scripts folder)

Reviewer: How about we mount the host folder that contains this script instead of instructing the user to create it?
Also, is it a python file or are you talking about `generate-rpm-specs.sh` here?

```
vim generate-junit-reports.sh
```
2.execute the bash file: bash generate-rpm-specs.sh <number of the specs>  . You will be able to see the RPM spec files in the folder
```
bash generate-rpm-specs.sh 100
```
3.execute
```
rpmbuild -bb *.spec
```

> You will see output like: Wrote: /root/rpmbuild/RPMS/x86_64/package_name9-0-0.x86_64.rpm Obviously this tells you where the rpm files are

Reviewer: ```suggestion
You will see output like: 
`Wrote: /root/rpmbuild/RPMS/x86_64/package_name9-0-0.x86_64.rpm`
This gives you the path where RPM files were generated to.
```

4.install all packages using the rpm files just created: yum localinstall <path to rpm files>/.*rpm  
```
yum localinstall /root/rpmbuild/RPMS/x86_64/.*rpm
```
5.check the running ubi containers by executing: docker container ls and get the container id of the container where you are currently installing the rpm packages

> 6.execute docker export [container id]>[tar name].tar (e.g docker export c8c57bb7e926>demoV1.tar) and unzip the tar ball to get the file system

Reviewer: What parts of the extracted filesystem are we interested in? Maybe we should tell the reader here.

> 7.Go to **github.com/stackrox/scanner/blob/master/benchmarks/analyzeNode** folder and make sure to add local path to file system in line 13, then execute command:

Reviewer: If it is a single point in the code where we provide this path, how about we either hardcode it and tell the user where to put the required files, or we use an environment variable?

# Steps to benchmark node scanning 

## Pre steps:

quay.io/repository/rhacs-eng/sandbox:benchmark-node-analyze-1.0 is an Ubi8 based image with vim, RPM and RPM-build installed. Those packages are necessary for benchmark node analyze function

> back-up image: `quay.io/rh_ee_yli3/nodes:benchmark-node-analyze-1.0`

Reviewer 2: What is the purpose of the backup image?

Should we move it from the `yli3` namespace into something generic?

execute
```
docker run -it --rm quay.io/repository/rhacs-eng/sandbox:benchmark-node-analyze-1.0
```
---------------------------------------------------------------------------------------------

(optional) create a temporary/target folder in the running ubi8 bash for RPM specs. e.g mkdir temp-specs

1.go to the target folder and create a bash script to generate RPM specs (see scanner/scripts/node/generate-rpm-specs.sh)
```
vim generate-junit-reports.sh
```
2.execute the bash file: bash generate-rpm-specs.sh <number of the RPM specs>  . You will be able to see the RPM spec files in the folder
```
bash generate-rpm-specs.sh 100
```
3.execute
```
rpmbuild -bb *.spec
```
You will see output like: `Wrote: /root/rpmbuild/RPMS/x86_64/package_name9-0-0.x86_64.rpm` This gives you the path where RPM files were generated to.

4.install all packages using the rpm files just created: yum localinstall <path to rpm files>/.*rpm  
```
yum localinstall /root/rpmbuild/RPMS/x86_64/.*rpm
```
5.check the running ubi containers by executing: `docker container ls` and get the container id of the container where you are currently installing the rpm packages

> 6.execute `docker export [container id]>[tar name].tar` (e.g `docker export c8c57bb7e926>demoV1.tar`) and unzip the tar ball to get the file system

Reviewer 2: Nitpick: Very unlikely to happen, but better to have a space between `[container id]` and `>`: if the ID is all numeric and small enough, the shell might believe you are referring to a file descriptor instead.

> 7.Go to **scanner/blob/master/benchmarks/analyzeNode** folder and make sure to add local path to file system in line 13, then execute command:

Reviewer 2: I believe you need [a space between `.` and the beginning of the list content](https://github.github.com/gfm/#lists).

Also, for all list items, I think the list contents are English sentences, and AFAIK, it would be better style to Capitalize the first letter.
